File: software/examples/csharp/ExampleInterrupt.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Industrial Digital In 4 Bricklet

	// Callback function for interrupt callback
	static void InterruptCB(BrickletIndustrialDigitalIn4 sender, int interruptMask,
	                        int valueMask)
	{
		Console.WriteLine("Interrupt Mask: " + Convert.ToString(interruptMask, 2));
		Console.WriteLine("Value Mask: " + Convert.ToString(valueMask, 2));
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIndustrialDigitalIn4 idi4 =
		  new BrickletIndustrialDigitalIn4(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register interrupt callback to function InterruptCB
		idi4.Interrupt += InterruptCB;

		// Enable interrupt on pin 0
		idi4.SetInterrupt(1 << 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/javascript/ExampleInterrupt.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Industrial Digital In 4 Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var idi4 = new Tinkerforge.BrickletIndustrialDigitalIn4(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Enable interrupt on pin 0
        idi4.setInterrupt(1 << 0);
    }
);

// Register interrupt callback
idi4.on(Tinkerforge.BrickletIndustrialDigitalIn4.CALLBACK_INTERRUPT,
    // Callback function for interrupt callback
    function (interruptMask, valueMask) {
        console.log('Interrupt Mask: ' + interruptMask.toString(2));
        console.log('Value Mask: ' + valueMask.toString(2));
        console.log();
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
